src/efi.rs

```diff
@@ -860,4 +1103,170 @@ Boot0003* test";
+                    "Package directory uboot/2023.04-2 should be created"
+                );
+
+                // Files should be copied to the EFI subdirectory
+                let efi_dir = uboot_dir.join("EFI");
+                assert!(efi_dir.exists(), "EFI directory should be created");
+
+                // Verify that u-boot.bin was copied to EFI subdirectory
+                let copied_uboot_bin = efi_dir.join("u-boot.bin");
+                assert!(copied_uboot_bin.exists(), "u-boot.bin should be copied");
+                let uboot_content = fs::read_to_string(&copied_uboot_bin)?;
+                assert_eq!(
+                    uboot_content, "uboot binary content",
+                    "u-boot.bin content should be preserved"
+                );
+
+                // Verify that overlays directory and i2c.dtb were copied to EFI subdirectory
+                let copied_overlays_dir = efi_dir.join("overlays");
+                assert!(
+                    copied_overlays_dir.exists(),
+                    "overlays directory should be copied"
+                );
+
+                let copied_overlay_dtb = copied_overlays_dir.join("i2c.dtb");
+                assert!(copied_overlay_dtb.exists(), "i2c.dtb should be copied");
+                let overlay_content = fs::read_to_string(&copied_overlay_dtb)?;
+                assert_eq!(
+                    overlay_content, "device tree overlay content",
+                    "i2c.dtb content should be preserved"
+                );
+
+                // Verify the EFI.json metadata
+                let metadata_file = uboot_dir.join("EFI.json");
+                assert!(
+                    metadata_file.exists(),
+                    "EFI.json metadata file should be created"
+                );
+                let metadata_content = fs::read_to_string(&metadata_file)?;
+                let parsed: ContentMetadata = serde_json::from_str(&metadata_content)?;
+                assert!(
+                    parsed
+                        .version
+                        .contains("uboot-images-2023.04-2.fc42.noarch"),
+                    "Metadata should contain uboot package"
+                );
+
+                println!("extend_payload test completed successfully!");
+                println!("✓ Files copied to: {:?}", efi_dir);
+                println!("✓ u-boot.bin: {:?}", copied_uboot_bin);
+                println!("✓ overlays/i2c.dtb: {:?}", copied_overlay_dtb);
+                println!("✓ Metadata created: {:?}", metadata_file);
+                println!("✓ Package version: {}", parsed.version);
+            }
+            Ok(Some(false)) => {
+                panic!("extend_payload returned false - expected success");
+            }
+            Ok(None) => {
+                panic!("extend_payload returned None - expected success");
+            }
+            Err(e) => {
+                panic!("extend_payload failed when it should have succeeded: {}", e);
```

Comment:
It'd avoid all this nesting to just use `.unwrap()` or `.expect()` on the result which seems preferable to me.
And we may as well do the same for the `Option`